File: JavaScript/프로그래머스/3/43162. 네트워크/네트워크.js
function dfs(n, computers, idx, visited) {
  visited[idx] = true;
  for (let i = 0; i < n; i++) {
    if (computers[idx][i] === 1 && !visited[i]) {
      dfs(n, computers, i, visited);
    }
  }
}

function solution(n, computers) {
  let answer = 0;
  const visited = Array.from({ length: n }, () => false);
  for (let i = 0; i < n; i++) {
    if (!visited[i]) {
      answer++;
      dfs(n, computers, i, visited);
    }
  }
  return answer;
}

File: C++/프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>

using namespace std;
void dfs(int n, vector<vector<int>> computers, int idx, vector<bool> &visited) {
  visited[idx] = true;
  for (int i = 0; i < n; i++) {
    if (computers[idx][i] == 1 && !visited[i]) {
      dfs(n, computers, i, visited);
    }
  }
}

int solution(int n, vector<vector<int>> computers) {
  int answer = 0;
  vector<bool> visited(n, false);
  for (int i = 0; i < n; i++) {
    if (!visited[i]) {
      answer++;
      dfs(n, computers, i, visited);
    }
  }
  return answer;
}
